pkg/skaffold/schema/defaults/defaults.go

```diff
@@ -210,6 +210,7 @@ func setDefaultClusterTimeout(cluster *latest.ClusterDetails) error {
 }
 
 func setDefaultClusterPullSecret(cluster *latest.ClusterDetails) error {
+	cluster.PullSecretMountPath = valueOrDefault(cluster.PullSecretMountPath, constants.DefaultKanikoSecretMountPath)
```

Comment:
kaniko secret can be optional now. In that case, we need to move this down i guess?

Reply:
This needs to be set for both `PullSecret` and `PullSecretName`, so I put it at the top. If neither of those are set, this will never be used, so it's ok to set it up here.

Reply:
umm, so we end up setting defaults for unused field.
Maybe, we should step back.. so, i looked at the bug you pointed, here. I will grab you offline.